rrdd/rrdd_monitor.ml

```diff
@@ -40,6 +40,17 @@ let merge_new_dss rrd dss =
       rrd_add_ds rrd now (Rrd.ds_create ds.ds_name ds.Ds.ds_type ~mrhb:300.0 Rrd.VT_Unknown)
     ) rrd new_dss
 
+module StringSet = Set.Make(String)
+
+let remove_unused_dss rrdi =
+  let all_dss = rrdi.rrd |> Rrd.ds_names in
+  let used_dss = rrdi.dss |> List.map (fun d -> d.ds_name) in
+  if List.length all_dss = List.length used_dss then rrdi
```

Comment:
Is this optimisation worth it? We need to travers at least both list. There is also the theoretical problem that both lists could have equal length but contain different names. The optimisation does save allocation of short-lived memory, though.

Reply:
Most of the time there is nothing to remove, so we can avoid building the stringsets.
IIUC the RRD would contain at least the entries from the DSS, possibly more, but perhaps we shouldn't rely on that detail here.